smoke-tests/run_smoke_test.sh

```diff
@@ -485,6 +487,26 @@ verify_output_empty "${output}" false
 
 finish_step
 
+###############################################
+#### Step 3b (temp) - Verify mount works   ####
+###############################################
+# Due to some issues with the github actions macOS executors, macfuse
+# doesn't work in our integration tests, so we ensure that mounts
+# in our smoke tests. The integration tests *do* test mounts on
+# Windows and Linux so this testing is really just to be extra cautious.
+# We can remove this whole step when/if the macfuse issue is cleared up
+# in the github actions executors
```

Comment:
do we have a reference of some sort here? maybe we should open up an issue with github to track this

Reply:
So perhaps I should change this comment... This issue was happening in the newer circle ci executors as well (which is why we were pinned to an old version) so I think it's actually due to changes in macos since it was affecting both.  And so once it seemed like a macos problem, I decided not to dig too much further since adding a smoke test would be an easy workaround... but I'll try to add more information here so people are aware of the "state" I'm leaving it in.